.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,128 @@
+      shell: pwsh
+
+    # Move and rename the plugin for installation
+    - name: Move and rename the plugin for installation
+      run: mv ./build/distributions/${{ inputs.plugin-name }}-*.zip ${{ inputs.plugin-name }}.zip
+      shell: bash
+
+    # Install the plugin, runs its start-script, and start the OpenSearch server
+    - name: Install Plugin into OpenSearch for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest'}}
+      run: |
+        cat > os-ep.sh <<EOF
+        yes | opensearch-plugin install file:///docker-host/${{ inputs.plugin-zip }}.zip
+        chmod +x plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        yes | plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        chown 1001:1001 -R /opensearch
+        su -c "/opensearch/bin/opensearch" -s /bin/bash opensearch
+        EOF
+        docker build -t opensearch-test:latest -f- . <<EOF
+        FROM ubuntu:latest
+        COPY --chown=1001:1001 os-ep.sh /docker-host/
+        COPY --chown=1001:1001 ${{ inputs.plugin-name }}.zip /docker-host/${{ inputs.plugin-zip }}.zip
+        COPY --chown=1001:1001 opensearch* /opensearch/
+        RUN chmod +x /docker-host/os-ep.sh
+        RUN useradd -u 1001 -s /sbin/nologin opensearch
+        ENV PATH="/opensearch/bin:${PATH}"
+        WORKDIR /opensearch/
+        ENTRYPOINT /docker-host/os-ep.sh
+        EOF
+      shell: bash
+
+    - name: Install Plugin into OpenSearch for Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: |
+        'y' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch-plugin.bat install file:$pwd\${{ inputs.plugin-name }}.zip
+        'y', 'y', 'N' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\plugins\${{ inputs.plugin-name }}\tools\${{ inputs.plugin-start-script }}.bat
+      shell: pwsh
+
+    # Run OpenSearch
+    - name: Run OpenSearch with plugin on Linux
+      if: ${{ inputs.platform == 'ubuntu-latest'}}
+      run: docker run --name ops -d -p 9200:9200 -p 9600:9600 -i opensearch-test:latest
+      shell: bash
+
+    - name: Run OpenSearch with plugin on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: start .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch.bat
+      shell: pwsh
+
+    # Give the OpenSearch process some time to boot up before sending any requires, might need to increase the default time!
+    - name: Sleep while OpenSearch starts
+      uses: peternied/action-sleep@v1
+      with:
+        seconds: 30
+
+    # Verify that the server is operational
+    - name: Check OpenSearch Running on Linux and MacOS
+      if: ${{ inputs.platform != 'windows-latest'}}
+      run: curl https://localhost:9200 -u 'admin:admin' -k -v
+      shell: bash
+
+    - name: Check OpenSearch Running on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: |
```

Comment:
Before the web request please cat the opensearch.log (or whatever its called) file for troubleshooting

Reply:
You mean to print the whole log file or to simply pipe the output of running opensearch into the file? I.e. do you mean cat the log or cat opensearch into the log?

Reply:
Print the whatever is in that log file so it can be read in the workflow log

Reply:
Seems like the log file is empty? Id this expected? I don't think it is.

Reply:
I don't think `logs\gc.log.00` is correct, that is a garbage collection log, I think it should be named opensearch.log in that same folder

Reply:
Yeah that is what I thought too but the only file with any bytes in the directory is the gc file. When I try to cat opensearch.log I get an error because it is empty. What would you like me to do?

Reply:
Made an issue because the log file is empty: https://github.com/opensearch-project/security/issues/2251